packages/toolkit/core-kit/src/password-policy.ts

```diff
@@ -282,171 +313,143 @@ export class PasswordPolicyChecker {
    */
-  hasWords(password: string): string[] {
-    const words = this.policy.rejects.words.map((word) => word.toLowerCase());
-    const lowercasedPassword = password.toLowerCase();
-    const threshold = PasswordPolicyChecker.getRestrictedPhraseThreshold(password);
+  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
+  wordLength(password: string): number {
+    const sliced = password.toLowerCase();
+    let length = 0;
+
+    for (const word of this.policy.rejects.words) {
+      if (sliced.startsWith(word.toLowerCase()) && word.length > length) {
+        length = word.length;
+      }
+    }
 
-    return words.filter((word) => word.length >= threshold && lowercasedPassword.includes(word));
+    return length;
   }
+  /* eslint-enable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
 
   /**
-   * Check if the given password contains sequential characters, and the number of
-   * sequential characters is over the threshold.
-   *
-   * @param password - Password to check.
-   * @returns Whether the password contains sequential characters.
+   * Get the length of the sequence at the beginning of the given string.
+   * For example, `sequenceLength('12345')` will return `5`.
    *
-   * @example
-   * ```ts
-   * hasSequentialChars('1'); // false
-   * hasSequentialChars('12'); // false
-   * hasSequentialChars('12345'); // true
-   * hasSequentialChars('123456@Bc.dcE'); // true
-   * hasSequentialChars('123@Bc.dcE'); // false
-   * ```
+   * If the length is less than {@link PasswordPolicyChecker.repetitionAndSequenceThreshold},
+   * `0` will be returned.
    */
-  // Disable the mutation rules because the algorithm is much easier to implement with
-  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation, @silverhand/fp/no-mutating-methods */
-  hasSequentialChars(password: string): boolean {
-    const { repetitionAndSequenceThreshold, getRestrictedPhraseThreshold } = PasswordPolicyChecker;
-    if (password.length < repetitionAndSequenceThreshold) {
-      return false;
-    }
-
-    const threshold = Math.max(
-      getRestrictedPhraseThreshold(password),
-      repetitionAndSequenceThreshold
-    );
-    let sequence: number[] = [];
+  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
+  sequenceLength(password: string): number {
+    const { repetitionAndSequenceThreshold } = PasswordPolicyChecker;
+    let value = '';
+    let length = 0;
 
     for (const char of password) {
-      // Always true
-      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-      const charCode = char.codePointAt(0)!;
-
-      if (sequence.length === 0) {
-        sequence.push(charCode);
-        continue;
-      }
-
-      if (
-        this.checkSequentialWithChar(sequence, charCode, 1) ||
-        this.checkSequentialWithChar(sequence, charCode, -1)
-      ) {
-        sequence.push(charCode);
+      if (!value || this.isSequential(value + char)) {
```

Comment:
nit:
```
if (XXX) {
 XXXXX;
 continue;
}
break;
```

Reply:
looks kind of weird to use break as the main expression in a for loop

Reply:
discussed offline